src/Search.hs

```diff
@@ -0,0 +1,54 @@
+module Search
+    ( searchBoards
+    ) where
+
+import Control.Monad (mzero)
+import Data.Maybe (fromMaybe)
+import Data.List (filter)
+
+import Data.Aeson
+import qualified Data.ByteString.Lazy as BS
+import Data.Text as T (Text, pack)
+import Text.XML.Generator
+import qualified Text.Fuzzy as Fuzzy
+
+data Board = Board
+    { boardName :: Text
+    , boardId :: Text
+    , boardUrl :: Text
+    } deriving (Show)
+
+instance FromJSON Board where
+    parseJSON (Object v) = Board
+                            <$> v .: "name"
+                            <*> v .: "id"
+                            <*> v .: "url"
+    parseJSON _          = mzero
+
+searchBoards :: Maybe String -> IO ()
+searchBoards mq = do
```

Comment:
I think you could have `matchingBoards` take a `String` instead of a `Maybe String` by pattern matching here:

```haskell
searchBoards Nothing = return ()
searchBoards (Just q) = do
    json <- BS.readFile "boards.json"
    let mboards = decode json
     printXML $ matchingBoards mboards q
```